leapp/utils/lock.py

```diff
@@ -0,0 +1,96 @@
+import os
+import fcntl
+import logging
+
+from leapp.config import get_config
+from leapp.exceptions import ProcessLockError
+
+
+def leapp_lock(lockfile=None):
+    return ProcessLock(lockfile=lockfile)
+
+
+def _acquire_lock(fd):
+    try:
+        fcntl.flock(fd, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        return True
+    except OSError:
+        return False
+
+
+def _read_pid(fd):
+    return os.read(fd, 20)
+
+
+def _write_pid(fd, pid):
+    os.write(fd, str(pid).encode('utf-8'))
+
+
+def _validate_pid(old_pid):
+    try:
+        old_pid = int(old_pid)
+    except ValueError:
+        return False
+
+    return old_pid
+
+
+def _clear_lock(fd):
+    os.lseek(fd, 0, os.SEEK_SET)
+    os.ftruncate(fd, 0)
+
+
+class ProcessLock(object):
+
+    def __init__(self, lockfile=None):
+        self.log = logging.getLogger('leapp.utils.lock')
+        self.lockfile = lockfile if lockfile else get_config().get('lock', 'path')
+
+        self.fd = None
+
+    def _get_pid_from_lockfile(self):
+        running_pid = _read_pid(self.fd)
+        self.log.debug("_get_pid_from_lockfile: running_pid=%s", running_pid)
+
+        if len(running_pid) == 0:
+            # No pid in lockfile
+            return
+
+        running_pid = _validate_pid(running_pid)
+        if not running_pid:
+            return
+
+        return running_pid
```

Comment:
```suggestion
    def _get_pid_from_lockfile(self):
        running_pid = _read_pid(self.fd)
        self.log.debug("_get_pid_from_lockfile: running_pid=%s", running_pid)
        running_pid = int(running_pid)
        
        return running_pid
```

It feels like the error checking and exception handling should be done by the caller of this function.  The reason is that `_get_pid_from_lockfile()` doesn't know what to do if the lockfile does not contain a pid.  Is it an error?  Should a warning be emitted?  The caller knows these things but not `_get_pid_from_lockfile()` so the errors should raise an exception that the caller can handle.

In this case, `int("")` and `int("not a pid")` will raise a `ValueError` so we can just let that propogate up.